### DISCUSSION TITLE
Decimal / F0049 Discussion Thread

### DISCUSSION THREAD

**Discussion by adesutherland** [adesutherland/CREXX#378] [Progress Updates]
I have done some pushes - the main thing is I have moved the ICU library to the icu directory. The decimal directory contains the beginnings of the plugin. I have not yet started linking the plugin to the interpreter.

**Comment by Peter-Jacob** [comment]
We must not forget:

1.	new register->fields required:
	- register->declength // allocated/available storage length
	- register->digits // number of assigned digits
2.	Memory allocations/management
	- digit size(s)
	- memory size (now 128 Bytes)
	- freeing decimal storage at the end of a crexx
3.	comparisons
	- DEC <-> Integer
	- DEC <-> Float
4. 	64-bit Integer Decimal conversion outstanding
5.	also the double to Decimal conversion 

Adjusting it is no big deal, as I put the stuff in macros.
